docs/pluggable-discovery-specification.md

```diff
@@ -377,3 +377,67 @@ myboard.upload_port.1.apples=40
 
 will match on both `pears=20, apples=30` and `pears=30, apples=40` but not `pears=20, apples=40`, in that sense each
 "set" of identification properties is independent from each other and cannot be mixed for port matching.
+
+#### Identification of board options
+
+The board identification can also identify [custom board options](platform-specification.md#custom-board-options) if the
+corresponding `upload_port` properties are defined:
+
+```
+BOARDNAME.menu.MENUNAME.MENUOPTION.upload_port.IDENTIFIER=Value
+BOARDNAME.menu.MENUNAME.MENUOPTION.upload_port.#.IDENTIFIER=Value
+```
+
```

Comment:
```suggestion

If multiple identification properties are associated within a set, all must match for the option to be identified.

```

Specify that association of multiple properties within a set is an AND (as opposed to OR).